tests/baselines/reference/objectKeysWithRecord.types

```diff
@@ -0,0 +1,26 @@
+=== tests/cases/compiler/objectKeysWithRecord.ts ===
+type K = 'foo' | 'bar'
+>K : K
+
+const record: Record<K, boolean> = { foo: true, bar: false };
+>record : Record<K, boolean>
+>{ foo: true, bar: false } : { foo: true; bar: false; }
+>foo : true
+>true : true
+>bar : false
+>false : false
+
+Object.keys(record).forEach((key: K) => {
+>Object.keys(record).forEach((key: K) => {}) : void
+>Object.keys(record).forEach : (callbackfn: (value: string, index: number, array: string[]) => void, thisArg?: any) => void
+>Object.keys(record) : string[]

```

Comment:
Doesn't look like this is using the right overload. Your code is only compiling because you can assign a function (i.e. `(key: K) => {}`) whose argument is a subset of type `A` to a function whose argument is `A`. For example, this is valid:
```typescript
const abc: (def: number | string) => void = (def: number) => {}
```

Reply:
I think the change should only be in `src`. `lib` is auto-generated from that folder.